spec/README.md

```diff
@@ -0,0 +1,172 @@
+# smoothing-pool spec v1
+
+This document describes the specification of the DAppNode smoothing pool oracle for Ethereum validators. The existing code in this repository complies with this specification, so this serves as a more detailed explanation of the oracle components so that anyone can implement its own version, arriving at the same results and being interoperable with any implementation respecting this spec. It should allow anyone to reconstruct the same merkle root, which acts as a summary of all the validator balances and their rewards that are monitored by the oracle.
+
+## Introduction
+
+In a few words, the main motivation for this smoothing pool is the following. A single Ethereum validator, which requires 32 Eth, of collateral proposes an average of 6 blocks a year, and this number decreases with the number of active validators in the network. The so-called *solo stakers* are individuals that usually run a single validator, which makes them propose very few blocks a year. Even in some cases, we can have a tail of unlucky validators that will propose even fewer blocks.
+
+On the other hand, rewards in Ethereum are divided into **consensus** and **execution** rewards. Consensus rewards are quite stable and are awarded to the validators every epoch (6ish minutes). However, execution rewards are the ones awarded to validators proposing blocks. Since block proposals are rare, and their reward has a very high variance, this makes the return of investment a year of a single validator not predictable and subject to luck. In other words, a lucky validator can have a very high return, but there will be always a long tail of unlucky validators with rewards way below the average.
+
+The smoothing pool aims to solve this problem, by creating a common pool to which all solo stakers give their rewards. Then, every given amount of time, these rewards are shared equally by all participants. This makes rewards more predictable for entities/solo-stakers running few validators, allowing them to average out or *smooth* the rewards they get. Note that entities running thousands of validators are not subject to this issue, because with that amount of validators statistics plays on their favor, so they internally smooth the variance.
+
+As an example, if `validator_a` proposes a block with `reward_a` and `validator_b` with `reward_b`, it is statistically better that instead of each getting `reward_a` and `reward_b` respectively, each one gets `(reward_a+reward_b)/2`. This oracle contains all the calculations of the rewards each participant validator can claim, for an arbitrary number of validators that keep joining and leaving the pool, with some even cheating. The oracle also summarizes the data in a merkle tree, where the root is stored onchain to keep all accounting decentralized and as open as possible.
+
```

Comment:
self note.

explain:
* claimable
* accumulated
* pending
* claimed,

and use some analogies.
* pending = unlocked in next proposal
* accumulated = rewards since you join.
* claimable = what you accumulated since what you allreadyclaimed.